src/main/java/spoon/support/compiler/jdt/ParentExiter.java

```diff
@@ -1187,7 +1187,7 @@ public void visitCtRecord(CtRecord recordType) {
 	public void visitCtRecordPattern(CtRecordPattern pattern) {
 		CtElement child = adjustIfLocalVariableToTypePattern(this.child);
 		if (child instanceof CtTypeReference<?> typeReference) {
-			pattern.setRecordType(typeReference);
```

Comment:
I wouldn't expect the child to be used again somewhere (otherwise you probably would need to clone the CtPattern too?). Where is this used again?